### PR TITLE
meta-mbl-apps:recipes-core: Add recipe for mbl-watchdog-init utility.

### PR DESCRIPTION
**Pull request by rwalton-arm**
This patch introduces a recipe to build the mbl-watchdog-init utility.
The package is added to the initramfs image.

This patch also modifies the initramfs-init script to run the
mbl-watchdog-init util before we mount the rootfs.

Jenkins: http://jenkins.mbed-linux.arm.com/job/rw-test/160/
LAVA: http://e120856-lin.cambridge.arm.com/results.php?image_version=custom&image_number=160&custom_version=rw-test&lava_query=&lava_owner=&failures=false&submitter=robwal02&sort=by_job

Notes: 
Merge after after https://github.com/ARMmbed/mbl-core/pull/219 
SRCREV needs to be updated before merging

### CODE REVIEW

File: meta-mbl-distro/recipes-core/mbl-initramfs-init/files/initramfs-init-script.sh [hunk 1]
```diff
@@ -27,6 +30,12 @@ then
     exec </dev/console >/dev/console 2>/dev/console
 fi
 
+# Initialise the hardware watchdog and set the timeout. We need to do this
+# in the initramfs, as the watchdog needs to be started before switching to the rootfs.
+printf "Setting hardware watchdog with device filename %s to timeout of %s seconds\n" \
+    "${MBL_WATCHDOG_DEVICE_FILENAME}" "${MBL_WATCHDOG_TIMEOUT_SECS}"
+mbl-watchdog-init --timeout ${MBL_WATCHDOG_TIMEOUT_SECS} --device ${MBL_WATCHDOG_DEVICE_FILENAME}
```
Comment (jonathanhaigh-arm) [inline meta-mbl-distro/recipes-core/mbl-initramfs-init/files/initramfs-init-script.sh]: I suggest double quotes around `${MBL_WATCHDOG_TIMEOUT_SECS}` and `${MBL_WATCHDOG_DEVICE_FILENAME}`, even if only for consistency with the previous line (I don't see why they would need quoting in the `printf` line but not this one).

File: meta-mbl-distro/recipes-core/mbl-initramfs-init/files/initramfs-init-script.sh [hunk 2]
```diff
@@ -13,6 +13,9 @@ LOG_MOUNT_OPTS="__REPLACE_ME_WITH_MBL_LOG_MOUNT_OPTS__"
 LOG_MOUNT_POINT="__REPLACE_ME_WITH_MBL_LOG_MOUNT_POINT__"
 BOOTFLAGS_DIR="__REPLACE_ME_WITH_MBL_BOOTFLAGS_DIR__"
 ROOTFS_LABEL_BASE="__REPLACE_ME_WITH_MBL_ROOT_LABEL__"
+MBL_WATCHDOG_TIMEOUT_SECS="__REPLACE_ME_WITH_MBL_WATCHDOG_TIMEOUT_SECS__"
```
[the remark this answers is not in the document]
Comment (rwalton-arm) [reply meta-mbl-distro/recipes-core/mbl-initramfs-init/files/initramfs-init-script.sh]: They're set in the machine.conf for each machine

[the remark this answers is not in the document]
Comment (jonathanhaigh-arm) [reply meta-mbl-distro/recipes-core/mbl-initramfs-init/files/initramfs-init-script.sh]: Okay, thanks. I was expecting those definitions to be added in this PR, but they were part of the PR to get systemd to prod the watchdog fb9464de63195125714c2dacdd00a684d6502464.

**Comment by rwalton-arm** [comment]
@jonathanhaigh-arm I have updated mbl-linked-repositories.conf to point to the mbl-core commit which added the watchdog-init utility. Could you please approve again?

**Pull request merged**